src/SkirtBrim.cpp

```diff
@@ -387,15 +387,22 @@ Polygons SkirtBrim::getFirstLayerOutline(const int extruder_nr /* = -1 */)
         const bool skirt_around_prime_tower_brim = storage.primeTower.enabled && global_settings.get<bool>("prime_tower_brim_enable");
         const bool include_prime_tower = ! skirt_around_prime_tower_brim; // include manually otherwise
 
-        const int skirt_height = global_settings.get<int>("skirt_height");
         first_layer_outline = Polygons();
-        for (int i_layer = layer_nr; i_layer <= skirt_height; ++i_layer)
+        int skirt_height = 0;
+        for (const auto& extruder : Application::getInstance().current_slice->scene.extruders)
         {
-            first_layer_outline =
-                first_layer_outline.unionPolygons
-                (
-                    storage.getLayerOutlines(i_layer, include_support, include_prime_tower, external_only, extruder_nr)
-                );
+            skirt_height = std::min(std::max(skirt_height, extruder.settings.get<int>("skirt_height")), static_cast<int>(storage.print_layer_count));
```

Comment:
From the documentation: 
```
* \param extruder_nr The extruder for which to get the outlines. -1 to include outliens for all extruders
```
We should thus only include the skirt height of all extruders when this variable is set to `-1`. Taking some inspiration from this function
https://github.com/Ultimaker/CuraEngine/blob/14bb2f2959c06def09be91ae264c9681c44e8100/src/sliceDataStorage.cpp#L285
I would suggest to add something like
```suggestion
            if (extruder_nr == -1 || extruder_nr == extruder.extruder_nr)
            {
                skirt_height = std::min(std::max(skirt_height, extruder.settings.get<int>("skirt_height")), static_cast<int>(storage.print_layer_count));
            }
```